nutils/debug_flags.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2020 Evalf
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+
+_env = os.getenv('NUTILS_DEBUG', '')
+
+_default = set(['sparse', 'lower'])
+_all = _default.union(['evalf'])
+
+_flags = _all if _env == 'all' \
+    else set(filter(None, _env.lower().split(':'))) | _default if __debug__ \
+    else set(filter(None, _env.lower().split(':')))
+assert _flags <= _all, 'Unknown debug flags: {}'.format(_flags - _all)
+
+for f in _all:
+  locals()[f] = f in _flags
```

Comment:
Do we prefer them as module attributes or attribute of a constant like `DEBUG_FLAGS`?

Reply:
My preference is module attributes, the way you have it now.